fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/models/ResourceFactoryImpl.java

```diff
@@ -150,6 +150,35 @@ public FedoraResource getContainer(final String transactionId, final FedoraId re
         }
     }
 
+    @Override
+    public String getInteractionModel(final String transactionId, final FedoraId fedoraId) {
+        if (fedoraId.isRepositoryRoot()) {
```

Comment:
The headers for the RepositoryRoot also indicate its interaction model. Is the potential performance optimization here worth the redundant logic?